src/main/java/org/dependencytrack/tasks/OSVDownloadTask.java

```diff
@@ -0,0 +1,282 @@
+                }
+            }
+        }
+        vuln.setSeverity(calculateOSVSeverity(advisory));
+        vuln.setCvssV2Vector(advisory.getCvssV2Vector());
+        vuln.setCvssV3Vector(advisory.getCvssV3Vector());
+        return vuln;
+    }
+
+    // calculate severity of vulnerability on priority-basis (database, ecosystem)
+    public Severity calculateOSVSeverity(OSVAdvisory advisory) {
+
+        // derive from database_specific cvss v3 vector if available
+        if(advisory.getCvssV3Vector() != null) {
+            Cvss cvss = Cvss.fromVector(advisory.getCvssV3Vector());
+            Score score = cvss.calculateScore();
+            return normalizedCvssV3Score(score.getBaseScore());
+        }
+        // derive from database_specific cvss v2 vector if available
+        if (advisory.getCvssV2Vector() != null) {
+            Cvss cvss = Cvss.fromVector(advisory.getCvssV2Vector());
+            Score score = cvss.calculateScore();
+            return normalizedCvssV2Score(score.getBaseScore());
+        }
+        // get database_specific severity string if available
+        if (advisory.getSeverity() != null) {
+            if (advisory.getSeverity().equalsIgnoreCase("CRITICAL")) {
+                return Severity.CRITICAL;
+            } else if (advisory.getSeverity().equalsIgnoreCase("HIGH")) {
+                return Severity.HIGH;
+            } else if (advisory.getSeverity().equalsIgnoreCase("MODERATE")) {
+                return Severity.MEDIUM;
+            } else if (advisory.getSeverity().equalsIgnoreCase("LOW")) {
+                return Severity.LOW;
+            }
+        }
+        // get largest ecosystem_specific severity from its affected packages
+        if (advisory.getVulnerabilities() != null) {
+            List<Integer> severityLevels = new ArrayList<>();
+            for (OSVVulnerability vuln : advisory.getVulnerabilities()) {
+                severityLevels.add(vuln.getSeverity().getLevel());
+            }
+            Collections.sort(severityLevels);
+            Collections.reverse(severityLevels);
+            return getSeverityByLevel(severityLevels.get(0));
+        }
+        return Severity.UNASSIGNED;
+    }
+
+    public Vulnerability.Source extractSource(String vulnId) {
+
+        final String sourceId = vulnId.split("-")[0];
+        switch (sourceId) {
+            case "GHSA": return Vulnerability.Source.GITHUB;
+            case "CVE": return Vulnerability.Source.NVD;
+            default: return Vulnerability.Source.GOOGLE;
+        }
+    }
+
+    public VulnerableSoftware mapVulnerabilityToVulnerableSoftware(final QueryManager qm, final OSVVulnerability vuln) throws MalformedPackageURLException {
+
+        String versionStartIncluding = vuln.getLowerVersionRange();
+        String versionEndExcluding = vuln.getUpperVersionRange();
+
+        if (vuln.getPurl() == null) return null;
+
+        PackageURL purl = new PackageURL(vuln.getPurl());
```

Comment:
OSV is providing invalid PURLs for scoped NPM components, causing lots of parsing errors right now

```
2022-06-30 20:07:57,627 INFO [OSVDownloadTask] Updating datasource with Google OSV advisories for ecosystem npm
2022-06-30 20:07:57,819 ERROR [OSVDownloadTask] Error while mapping vulnerable software pkg:npm/@hapi/hoek
com.github.packageurl.MalformedPackageURLException: The PackageURL name specified is invalid
	at com.github.packageurl.PackageURL.validateName(PackageURL.java:280)
	at com.github.packageurl.PackageURL.parse(PackageURL.java:541)
	at com.github.packageurl.PackageURL.<init>(PackageURL.java:68)
	at org.dependencytrack.tasks.OSVDownloadTask.mapVulnerabilityToVulnerableSoftware(OSVDownloadTask.java:241)
	at org.dependencytrack.tasks.OSVDownloadTask.updateDatasource(OSVDownloadTask.java:130)
	at org.dependencytrack.tasks.OSVDownloadTask.unzipFolder(OSVDownloadTask.java:102)
	at org.dependencytrack.tasks.OSVDownloadTask.inform(OSVDownloadTask.java:75)
```

I raised an issue for OSV here: https://github.com/google/osv/issues/473

For this PR though, we should handle parsing errors here. An invalid PURL  should be treated as if no PURL was provided at all (just like the `null` check above).